### PR TITLE
Promoted istioctl x uninstall to istioctl uninstall

### LINKED ISSUE #40339: Experimental uninstall should be promoted to uninstall
**Describe the feature request**
Currently istio uninstall is an experimental command, over the times, it has been improved. However, it remains an experimental command and not being used by formal Istio document. Which creates some rather strange processes for example, when one use `istioctl install` to install istio, one has to use generate manifest file to uninstall istio, in many cases, the generated manifest file does not match istioctl install command created resources in the cluster, this problem caused many Istio issues.
 
**Describe alternatives you've considered**
promote experimental uninstall command to real istioctl commad

**Affected product area (please put an X in all that apply)**

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by litong01**
fixes #40339

Signed-off-by: Tong Li <litong01@us.ibm.com>

**Please provide a description of this PR:**

### CODE REVIEW

File: releasenotes/notes/40339.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 40339
+
+releaseNotes:
+- |
+  **Added** `istioctl x uninstall` promoted to `istioctl uninstall`.
```
Comment (ericvn) [inline releasenotes/notes/40339.yaml:9]: **Promoted** is a keyword.

```suggestion
  **Promoted** `istioctl experimental uninstall` to `istioctl uninstall`.
```

**Review by ericvn — COMMENTED** [review]
RN comment

File: releasenotes/notes/40339.yaml (hunk 1 above)
Comment (litong01) [reply releasenotes/notes/40339.yaml:9]: will change. thanks @ericvn

Comment (litong01) [reply releasenotes/notes/40339.yaml]: @ericvn done with the 2nd commit. Thanks.

File: releasenotes/notes/40339.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 40339
+
+releaseNotes:
+- |
+  **Promoted** `istioctl x uninstall` promoted to `istioctl uninstall`.
```
Comment (ericvn) [inline releasenotes/notes/40339.yaml:9]: Remove second promoted.

Comment (litong01) [reply releasenotes/notes/40339.yaml]: done @ericvn

**Review by ericvn — APPROVED** [review]
Release notes look good

**Review by GregHanson — APPROVED** [review]

**Comment by ericvn** [comment]
I'm thinking we should mark the experimental command deprecated and then remove it in the next minor release to give people time to move their scripts over. That's something the UX team likely has input/rules for.

**Comment by litong01** [comment]
> I'm thinking we should mark the experimental command deprecated and then remove it in the next minor release to give people time to move their scripts over. That's something the UX team likely has input/rules for.

@ericvn I've talked to @esnible and used softGraduatedCmd method, please see the latest commit.

**Review by therealmitchconnors — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Pull request merged**